File: Cpp/eu003.cpp
#include <iostream>
#include <vector>

int main() {
    long n = 600851475143;
    std::vector<int> primes({2});

    for (int x = 3; x != n; x += 2) {
        for(auto prime = primes.begin(); prime != primes.end(); ++prime) {
            if (x % (*prime) == 0) goto next;
        }
        primes.push_back(x);
        if (n % x == 0) n /= x;
        next:;
    }

    std::cout << n << std::endl;
}

File: Python/eu003.py
def primes():
    primes = [2]
    n = 1
    x = 3
    while True:
        for prime in primes:
            if x % prime == 0: break
        else:
            primes.append(x)
            yield x
        x += 2

x = 600851475143
for prime in primes():
    if x == prime:
        print(x)
        break
    if x % prime == 0:
        x /= prime
